internal/client/client.go

```diff
@@ -86,9 +86,26 @@ func (c *Config) Validate() error {
 	return nil
 }
 
+type ShutdownHandlerType int
+
+const ProxyShutdownHandler ShutdownHandlerType = iota
+
+func executeShutdownHandlers(
+	ctx context.Context,
+	m map[ShutdownHandlerType]func(ctx context.Context) error) error {
+	for _, f := range m {
+		err := f(ctx)
+		if err != nil {
+			return err
```

Comment:
I think we should accumulate errors, but also continue to the shutdown process. If something goes wrong, we the only action I can think of is logging. So would you find it useful to accumulate errors and continue to iterating?

Reply:
Also how about adding ShutdownHandlerType to the error for better context? In that case we may need to somehow convert to a string I guess.

Reply:
I created a custom error type for handler but, I'm not sure that I accumulate correctly [0dce062](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/0dce06299b7e0f4ec33083c83fbdf45de299988a).